subsys/bluetooth/mesh/sensor_cli.c

```diff
@@ -242,17 +242,24 @@ static int handle_column_status(struct bt_mesh_model *model, struct bt_mesh_msg_
 	struct bt_mesh_sensor_series_entry entry;
 
 	col_format = bt_mesh_sensor_column_format_get(type);
-
-	err = parse_series_entry(type, col_format, buf, &entry);
-	if (err == -ENOENT) {
-		/* The entry doesn't exist */
-		goto yield_ack;
+	
+	size_t val_len = sensor_value_len(type);
+	uint8_t count = buf->len / val_len;
+	if (col_format != NULL) {
+		val_len += (col_format->size * 2);
 	}
 
-	if (err) {
-		return err; /* Invalid format, should ignore message */
-	}
+	for (uint8_t i = 0; i < count; i++) {
+		err = parse_series_entry(type, col_format, buf, &entry);
+		if (err == -ENOENT) {
+			/* The entry doesn't exist */
+			goto yield_ack;
+		}
 
+		if (err) {
+			return err; /* Invalid format, should ignore message */
+		}
+	}
 	if (cli->cb && cli->cb->series_entry) {
 		cli->cb->series_entry(cli, ctx, type, 0, 1, &entry);
```

Comment:
Won't this be called only for the last entry from the loop above?

Reply:
yes, I think you're right, `cb` shall be called for every `entry`